internal/pkg/cli/deploy/workload.go

```diff
@@ -339,19 +339,27 @@ func (img ContainerImageIdentifier) customOrGitTag() string {
 	return img.GitShortCommitTag
 }
 
+// sidecarReferenceTag returns the tag that should be used to reference the image.
+func (img ContainerImageIdentifier) sidecarReferenceTag() string {
+	if img.GitShortCommitTag != "" {
+		return img.GitShortCommitTag
+	}
+	return img.uuidTag
+}
+
 func (d *workloadDeployer) uploadContainerImages(imgBuilderPusher imageBuilderPusher) (map[string]ContainerImageIdentifier, error) {
 	required, err := manifest.DockerfileBuildRequired(d.mft)
```

Comment:
I am thinking  of removing `BuildRequired() `from all the workloads as we can just depend on just  length of `BuildArgs` to determine docker build  required or not. what you folks think of it 🤔 .